pkg/garbagecollector/announcement_handlers.go

```diff
@@ -0,0 +1,74 @@
+package garbagecollector
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/rs/zerolog/log"
+
+	"github.com/openservicemesh/osm/pkg/announcements"
+	"github.com/openservicemesh/osm/pkg/constants"
+	"github.com/openservicemesh/osm/pkg/kubernetes/events"
+)
+
+// PatchSecretHandler patches the envoy bootstrap config secrets based on the PodAdd events
+// returns a stop channel which can be used to stop the inner handler
+func PatchSecretHandler(kubeClient kubernetes.Interface) chan struct{} {
+	podAddSubscription := events.GetPubSubInstance().Subscribe(announcements.PodAdded)
+	stop := make(chan struct{})
+
+	go func() {
+		for {
+			select {
+			case <-stop:
+				return
+			case podAddedMsg := <-podAddSubscription:
+				psubMessage, castOk := podAddedMsg.(events.PubSubMessage)
+				if !castOk {
+					log.Error().Msgf("Error casting PubSubMessage: %v", psubMessage)
```

Comment:
One trick I like to do in these cases is use `%T` in the format string to also log the type name to see what it actually is since sometimes that's not obvious from what `%v` gives. Something like:
```suggestion
					log.Error().Msgf("Error casting PubSubMessage: %T %v", psubMessage, psubMessage)
```